agent/roles/manifests/tasks/main.yml

```diff
@@ -18,6 +18,10 @@
   import_tasks: install-config.yml
   when: agent_use_ztp_manifests == 'false'
 
+- name: Patch agent-config manifest to test agent-tui
+  ansible.builtin.command: yq -i -y '.hosts[0].networkConfig["dns-resolver"].config.server[0] = "192.168.123.1"' "{{ install_path }}/agent-config.yaml"
```

Comment:
This works, thus I was expecting a change on the same template mechanism adopted for generating the config file. IINW the dns section is configured [here](https://github.com/openshift-metal3/dev-scripts/blob/122d238613e66e040ef049a609e9450ee84cb868/agent/roles/manifests/templates/agent-config_yaml.j2#L24), and the `provisioning_host_external_ip` is currently defined in [vars/main.yml](https://github.com/openshift-metal3/dev-scripts/blob/122d238613e66e040ef049a609e9450ee84cb868/agent/roles/manifests/vars/main.yml#L30).
Maybe a new `dns_resolver_ip` var could be added (and used for the templating), conditionally defined based on the `AGENT_TEST_CASES`, ie something like:

```
dns_resolver_ip: "{{ '192.168.123.1' if lookup('env', 'AGENT_TEST_CASES') == 'bad_dns' else provisioning_host_external_ip }}"
```
(not tested, but in the past I've used a similar approach)?

Reply:
Yes, I think @andfasano suggestion is good. For consistency/readability you could also add a new var in [here](https://github.com/openshift-metal3/dev-scripts/blob/master/agent/roles/manifests/vars/main.yml#L7) such as
agent_test_cases: lookup('env', 'AGENT_TEST_CASES')

and then use agent_test_cases == 'bad_dns' in the snippet above

Reply:
I've replaced the builtin command with conditionals in the template.